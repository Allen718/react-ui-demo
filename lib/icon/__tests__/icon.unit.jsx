import * as renderer from "react-test-renderer";
import React from "react";
import Icon from "../icon";
import {shallow} from 'enzyme'

describe("icon", () => {
  it("xxx", () => {
    const json = renderer.create(<Icon name="wechat" />).toJSON();
    expect(json).toMatchSnapshot();
  });
  it("可点击", () => {
    const fn = jest.fn();
    const component = shallow(<Icon onClick={fn} />);
    component.find("svg").simulate("click");
    expect(fn).toBeCalled();
  });
});
