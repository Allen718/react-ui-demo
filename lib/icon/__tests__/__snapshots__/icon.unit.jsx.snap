// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon xxx 1`] = `
<svg
  aria-hidden="true"
  className="icon"
>
  <use
    xlinkHref="wechat"
  />
</svg>
`;
